test/stage1/behavior/optional.zig

```diff
@@ -32,6 +32,39 @@ fn testNullPtrsEql() void {
     expect(&number == x);
 }
 
+test "equality compare optional with non-optional" {
+    test_cmp_optional_non_optional();
+    comptime test_cmp_optional_non_optional();
+}
+
+fn test_cmp_optional_non_optional() void {
```

Comment:
The test cases look good. Can you also add coverage for optional pointers (the motivating use case)?